packages/core/src/components/slider/multiSlider.tsx

```diff
@@ -228,78 +227,42 @@ export class MultiSlider extends AbstractPureComponent<IMultiSliderProps, ISlide
             const offsetPercentage = formatPercentage(offsetRatio);
             const style = this.props.vertical ? { bottom: offsetPercentage } : { left: offsetPercentage };
             labels.push(
-                <div className={`${Classes.SLIDER}-label`} key={i} style={style}>
+                <div className={Classes.SLIDER_LABEL} key={i} style={style}>
                     {this.formatLabel(i)}
                 </div>,
             );
         }
-        return <div className={`${Classes.SLIDER}-axis`}>{labels}</div>;
+        return labels;
     }
-    private renderFill() {
-        const trackStops = getSortedHandleProps(this.props);
-        trackStops.sort((left, right) => left.value - right.value);
-        if (trackStops.length === 0 || trackStops[0].value > this.props.min) {
-            trackStops.unshift({ value: this.props.min });
-        }
-        if (trackStops[trackStops.length - 1].value < this.props.max) {
-            trackStops.push({ value: this.props.max });
-        }
 
-        const tracks: Array<JSX.Element | null> = [];
-
-        for (let index = 0; index < trackStops.length - 1; index++) {
-            const left = trackStops[index];
-            const right = trackStops[index + 1];
-            const fillIntentPriorities = [
-                this.props.showTrackFill ? undefined : Intent.NONE,
-                left.intentAfter,
-                right.intentBefore,
-                this.props.defaultTrackIntent,
-            ];
-            const fillIntent = fillIntentPriorities.filter(intent => intent != null)[0];
-            tracks.push(this.renderTrackFill(index, left, right, fillIntent));
+    private renderTracks() {
+        const trackStops = getSortedHandleProps(this.props);
+        trackStops.push({ value: this.props.max });
+
+        // render from current to previous, then increment previous
+        let previous: ISliderHandleProps = { value: this.props.min };
```

Comment:
Does this still correctly handle non-interactive handles outside the slider range? Specifically, we had what we had previously to deal with cases like this:

```typescript
<Slider min={5} max={10} value={7} showTrackFill={true} />
```

Reply:
the `clamp()` in getOffsetRatio ensures segments must be between 0 and 1.

Reply:
added test case to be usre